.gitlab-ci.yml

```diff
@@ -379,32 +379,40 @@ test_acceptance_raspberrypi4_bookworm_64bit:
 
 .template:publish:s3:
   stage: publish
+  needs:
+    - job: convert_raspbian_raspberrypi4_bookworm_64bit
+      artifacts: true
+    - job: convert_raspbian_raspberrypi4_bookworm_32bit
+      artifacts: true
+    - job: convert_raspbian_raspberrypi4_bullseye_64bit
+      artifacts: true
+    - job: convert_raspbian_raspberrypi4_bullseye_32bit
+      artifacts: true
   image: debian:buster
   before_script:
     - apt update && apt install -yyq awscli curl
     - eval "$(curl https://raw.githubusercontent.com/mendersoftware/mendertesting/master/mender-ci-common.sh)"
-    # Fetch artifacts from temporary S3 bucket
+    - IMAGE_VERSION=${MENDER_CONVERT_PUBLISH_VERSION:-${CI_COMMIT_REF_NAME}}
+  script:
+    # One by one, fetch artifacts from temporary S3 bucket and publish to production S3 bucket
     - for RASPBERRYPI_CONFIG_NAME in raspberrypi4_bookworm_64bit raspberrypi4_bookworm_32bit raspberrypi4_bullseye_64bit raspberrypi4_bullseye_32bit; do
     -   mender_ci_load_tmp_artifact ${RASPBERRYPI_CONFIG_NAME}.tar.gz
     -   tar xzf ${RASPBERRYPI_CONFIG_NAME}.tar.gz
-    - done
-  script:
-    - IMAGE_VERSION=${MENDER_CONVERT_PUBLISH_VERSION:-${CI_COMMIT_REF_NAME}}
-
-    - for RASPBERRYPI_CONFIG_NAME in raspberrypi4_bookworm_64bit raspberrypi4_bookworm_32bit raspberrypi4_bullseye_64bit raspberrypi4_bullseye_32bit; do
     -   PUBLISH_NAME=${RASPBIAN_NAME_PUBLISH}-${RASPBERRYPI_CONFIG_NAME}-mender-convert-${IMAGE_VERSION}.img.xz
```

Comment:
Looking at this with fresh eyes,

This is wrong, isn't it? For the manual publish it will expand to: `2024-07-04-raspios-bookworm-arm64-lite-${RASPBERRYPI_CONFIG_NAME}-mender-convert-1.2.3`, with ambiguous cases like `2024-07-04-raspios-bookworm-arm64-lite-raspberrypi4_bullseye_32bit-mender-convert-1.2.3`. And for the automatic release I have no idea what will happen.

I'll try a fix on this too :+1: